arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/Schedule.kt

```diff
@@ -192,21 +212,48 @@ sealed class Schedule<Input, Output> {
   /**
    * Conditionally checks on both the input and the output whether or not to continue.
    */
-  abstract fun <A : Input> check(pred: suspend (A, Output) -> Boolean): Schedule<A, Output>
+  abstract fun <A : Input> check(pred: suspend (input: A, output: Output) -> Boolean): Schedule<A, Output>
 
   /**
    * Inverts the decision of a schedule.
    */
   abstract operator fun not(): Schedule<Input, Output>
 
-  /**
-   * Combines with another schedule by combining the result and the delay of the [Decision] with the functions [f] and [g]
-   */
-  abstract fun <A : Input, B> combineWith(
+  @Deprecated(
+    "combineWith is deprecated in favor of combineNanos and combine using Double/kotlin.time.Duration instead of arrow.fx.coroutines.Duration.",
```

Comment:
Also, I'd suggest aligning the deprecated messages if possible:

- "is deprecated in favor of"
- "will be replaced by"
- "is renamed to"